src/tests/integration_tests/test_entity_citation_validator.py

```diff
@@ -0,0 +1,99 @@
+from unittest.mock import patch
+
+import pytest
+from loguru import logger
+from nltk.metrics import jaccard_distance
+
+import sherpa_ai.config as cfg
+from sherpa_ai.agents.qa_agent import QAAgent
+from sherpa_ai.events import EventType
+from sherpa_ai.memory import SharedMemory
+from sherpa_ai.models.sherpa_base_chat_model import SherpaChatOpenAI
+from sherpa_ai.output_parsers.entity_validation import EntityValidation
+
+# from sherpa_ai.test_utils.llms import get_llm
+from sherpa_ai.tools import SearchTool
+from sherpa_ai.utils import extract_entities
+
+
+@pytest.mark.parametrize(
+    "objective , input_data, expected_entities",
```

Comment:
make the order of parameters the same as the function arguments. Currently, it's very hard to understand the test cases